### PR TITLE
[Core] `is_cosxl_edit` arg in SDXL ip2p.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

As found out by @yiyixuxu here:
https://github.com/huggingface/diffusers/issues/7621#issuecomment-2046997212

Related to https://github.com/huggingface/diffusers/issues/7621. 

Cc: @apolinario for awareness.

### CODE REVIEW

File: src/diffusers/loaders/single_file.py [hunk 1]
```diff
@@ -147,6 +148,8 @@ def set_additional_components(
                 "force_zeros_for_empty_prompt": False if is_refiner else True,
             }
         )
+    if pipeline_class_name == "StableDiffusionXLInstructPix2PixPipeline":
```
Comment (DN6) [inline src/diffusers/loaders/single_file.py:151]: Not needed. The kwarg will get passed and set in the pipeline here
https://github.com/huggingface/diffusers/blob/08bf7545070764d41637e90290ca9b92b392263e/src/diffusers/loaders/single_file.py#L317

Comment (sayakpaul) [reply src/diffusers/loaders/single_file.py]: Addressed in 56ad342f

File: src/diffusers/loaders/single_file.py [hunk 2]
```diff
@@ -252,6 +252,7 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
         local_files_only = kwargs.pop("local_files_only", False)
         revision = kwargs.pop("revision", None)
         torch_dtype = kwargs.pop("torch_dtype", None)
+        is_cosxl_edit = kwargs.pop("is_cosxl_edit", None)
```
Comment (DN6) [inline src/diffusers/loaders/single_file.py:255]: Not needed. 
https://github.com/huggingface/diffusers/blob/08bf7545070764d41637e90290ca9b92b392263e/src/diffusers/loaders/single_file.py#L317

File: src/diffusers/loaders/single_file.py [hunk 3]
```diff
@@ -309,7 +310,7 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
                 init_kwargs.update(components)
 
         additional_components = set_additional_components(
-            class_name, original_config, checkpoint=checkpoint, model_type=model_type
+            class_name, original_config, checkpoint=checkpoint, model_type=model_type, is_cosxl_edit=is_cosxl_edit
```
Comment (DN6) [inline src/diffusers/loaders/single_file.py:313]: Not needed. 
https://github.com/huggingface/diffusers/blob/08bf7545070764d41637e90290ca9b92b392263e/src/diffusers/loaders/single_file.py#L317

**Comment by sayakpaul** [comment]
@DN6 your comments have been addressed.

**Comment by sayakpaul** [comment]
Failing test is quite unrelated.

**Comment by sayakpaul** [comment]
Here's testing code:

```python
import torch
from diffusers import StableDiffusionXLInstructPix2PixPipeline, EDMEulerScheduler
from diffusers.utils import load_image
import numpy as np
import math

def set_timesteps_patched(self, num_inference_steps: int, device = None):
    self.num_inference_steps = num_inference_steps
    
    ramp = np.linspace(0, 1, self.num_inference_steps)
    sigmas = torch.linspace(math.log(self.config.sigma_min), math.log(self.config.sigma_max), len(ramp)).exp().flip(0)
    
    sigmas = (sigmas).to(dtype=torch.float32, device=device)
    self.timesteps = self.precondition_noise(sigmas)
    
    self.sigmas = torch.cat([sigmas, torch.zeros(1, device=sigmas.device)])
    self._step_index = None
    self._begin_index = None
    self.sigmas = self.sigmas.to("cpu")  # to avoid too much CPU/GPU communication

inst_file = "https://huggingface.co/stabilityai/cosxl/blob/main/cosxl_edit.safetensors"

pipe = StableDiffusionXLInstructPix2PixPipeline.from_single_file(
    inst_file, num_in_channels=8, is_cosxl_edit=True
).to("cuda")

# This won't be needed after https://github.com/huggingface/diffusers/pull/7649
EDMEulerScheduler.set_timesteps = set_timesteps_patched
pipe.scheduler = EDMEulerScheduler(sigma_min=0.002, sigma_max=120.0, sigma_data=1.0, prediction_type="v_prediction")

resolution = 1024
image = load_image(
    "https://hf.co/datasets/diffusers/diffusers-images-docs/resolve/main/mountain.png"
).resize((resolution, resolution))

edit_instruction = "Turn sky into a cloudy one"
edited_image = pipe(
    prompt=edit_instruction,
    image=image,
    height=resolution,
    width=resolution,
    num_inference_steps=20,
).images[0]
edited_image.save("edited_image.png")
```

^ matches what Yiyi got here: https://github.com/huggingface/diffusers/issues/7621#issuecomment-2046997212.

**Comment by sayakpaul** [comment]
Created an issue for the installation failure here: https://github.com/astral-sh/uv/issues/3031. 

I will submit a quick PR replacing the `uv` installation step that's causing the failure to unblock this PR.

**Review by DN6 — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**